src/components/Database/DataViewer/common/fetch.ts

```diff
@@ -89,17 +89,25 @@ function realtimeToViewModel(
   ref: firebase.database.Reference,
   queryParams: QueryParams
 ) {
-  const query = applyQuery(ref, queryParams);
+  let query: firebase.database.Query;
```

Comment:
I reverted the `once(query)` because it would have yielded strange results..

The check is really: 

does the **NODE** have children?
- if so, apply query
- else, it's a leaf so no queries.

If we left it as `once(query)` it would have done bad things:

Does the query have children?
- yes: all good
- no: return the full unfiltered node -- this would mean that queries that yield no results magically return everything.